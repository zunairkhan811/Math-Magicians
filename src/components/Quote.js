import { useState, useEffect } from 'react';

const Quote = () => {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);
  const [data, setData] = useState('');

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=humor', {
          method: 'GET',
          headers: {
            'Content-Type': 'application/json',
            'X-Api-key': 'caQKmJfAffKkic4Ds+HCAg==mQxfwIuFGKHRqdIW',
          },
        });
        const newdata = await response.json();
        setData(newdata);
        setLoading(false);
      } catch (error) {
        setLoading(false);
        setError(error.message);
      }
    };
    fetchData();
  }, []);

  if (loading) return <div className="quote"><p>Loading ...</p></div>;
  if (error) return <div className="quote"><p>Sorry cannot Load Data!!</p></div>;
  return (
    <>
      <div className="quote">
        <p>
          Quote:
          {' '}
          {data[0].quote}
          <br />
          By:
          {' '}
          {data[0].author}
        </p>
      </div>
    </>
  );
};

export default Quote;
