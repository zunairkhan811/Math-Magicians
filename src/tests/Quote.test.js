import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../components/Quote';

test('renders Quote component correctly', () => {
  const { asFragment } = render(<Quote />);
  expect(asFragment()).toMatchSnapshot();
});
