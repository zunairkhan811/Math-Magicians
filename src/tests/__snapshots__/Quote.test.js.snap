// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Quote component correctly 1`] = `
<DocumentFragment>
  <div
    class="quote"
  >
    <p>
      Loading ...
    </p>
  </div>
</DocumentFragment>
`;
